### DISCUSSION TITLE
Support for Intel NVMe driver Library

### DISCUSSION THREAD

**Discussion by alt-savage** [madMAx43v3r/chia-plotter#859] [Ideas]
Intel has developed a high performance,  zero-copy, Linux C library for directly interfacing with NVMe drives:

https://spdk.io/doc/nvme.html

Could this speedup Madmax NVMe  drive IO ?

**Comment by gryan315** [comment]
I asked this a few weeks ago, based on my research into the SPDK project as a non-developer, it appears to me that there is no way to use the driver as a universal driver for any application. The driver seems to only be usable by a custom application created specifically to use it, meaning you can't just set up a program to call up the SPDK library to run external programs. Mad max would have to be rewritten to include SPDK functionality, and would likely only benefit users using extremely high core count CPUs and large NVMe arrays.
